src/Cli/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -148,14 +177,39 @@ public List<string> Convert(string[] args, out List<string> ignoredArgs)
                     {
                         UpdateVerbosity(arg, newArgList);
                     }
-                    else if (IgnoredArguments.Contains(activeArgument))
+                    // This case should be processed before 's_ignoredArguments.Contains'.
+                    else if (s_switchArguments.Contains(activeArgument))
+                    {
+                        if (s_ignoredArguments.Contains(activeArgument))
+                        {
+                            ignoredArgs.Add(activeArgument);
+                        }
+                        else
+                        {
+                            newArgList.Add(activeArgument);
+                        }
+
+                        // The only real case where we would end-up here is when the item under test (.dll or .exe)
+                        // is passed as last argument so we could potentially check the extension and for invalid ones,
+                        // either add it to the ignored args or have a specific failure.
+                        newArgList.Add(arg);
+                    }
+                    else if (s_ignoredArguments.Contains(activeArgument))
                     {
                         ignoredArgs.Add(activeArgument);
                         ignoredArgs.Add(arg);
                     }
-                    else if (blame.IsBlameArg(activeArgument, arg))
+                    else if (BlameArgs.IsBlameArg(activeArgument))
                     {
-                        blame.UpdateBlame(activeArgument, arg);
+                        if (!BlameArgs.IsBlameSwitch(activeArgument)
```

Comment:
```suggestion
                        // We know that activeArgument is a blame kind, we now want to see if arg is linked to the
                        // blame or not. 
                        // If activeArgument is a not blame switch (e.g., --blame-crash-dump-type full) or if it is the
                        // legacy blame syntax (e.g., --blame CollectDump;DumpType=full) then we do want to 
                        // update the blame combination based both on activeArgument and arg.
                        // Otherwise, we have a simple blame switch (e.g., --blame some.dll) so we can add it to the
                        // args list.
```